### PR TITLE
Remove Kconfig dependency modelling

### PR DESCRIPTION
**Pull request by MrKevinWeiss**
### Contribution description

As discussed on [the forum](https://forum.riot-os.org/t/last-chance-for-kconfig-dependency-modelling/4068) and agreed upon in the VMA among other forms of communications we are removing the dependency modelling of Kconfig.  As it was never part of the "production" RIOT and only available as an experimental feature (`TEST_KCONFIG=1`) I do not think we need to deprecate it.

This PR should remove everything dependency related but preserve all "kconfig for configurations" aspects.

It is a step in the direction of having "one source of truth" for configurations, many configurations that were only exposed when `TEST_KCONFIG=1` are now exposed in a regular build (ie, `depends on MODULE_*` -> `depends on  USEMODULE_`).  These configurations are still not being used by default (for any test that does not have an app.config file present) as to not slow down the CI before the release but will be switched very soon.

There are quite some things needed in order to do this right:
- [x] Remove all obvious files that only are related to dep modelling (ideally just a script)
- [x] Remove all `MODULE_` and `PACKAGE_` configs and symbol references within remaining files
- [x] Alter all implemented `CONFIG_` that have been implemented under the dep modelling to use the make dep modelling (ie. `MODULE_` -> `USEMODULE_`)
- [x] Do manual fixes and alterations as needed
- [x] Cleanup any source issues for files that might be removed
- [x] Clean up the make system to remove all unneeded logic
- [x] Update the documentation to indicate the current state
- ~~Inject all the HAS_CPU* type features~~

### Testing procedure

Green murdock and a lot of looking through

### Issues/PRs references

Needs #20497

### CODE REVIEW

**Comment by MrKevinWeiss** [comment]
Hmm with should run kconfig removed, this might make benchmarking a bit harder...

**Review by benpicco — APPROVED** [review]
You probably want so squash that

**Comment by MrKevinWeiss** [comment]
Some tests are still failing, ya I will squash it... I guess it would be near impossible to review.

**Comment by MrKevinWeiss** [comment]
@benpicco Ok, I squashed the best I could... Still not so easy to review but if all tests are passing I guess it should be OK :wink: 

@Teufelchen1 A bit after soft feature freeze but I hope it is OK.  Keeping up with rebasing can be a bit of a challenge.

File: Kconfig [hunk 1]
```diff
@@ -41,23 +35,5 @@ menu "External Packages"
...
-        modelling. Don't change the default value unless you know what you are
-        doing.
-
-config DEVELHELP
-    bool "Development Help"
-    help
-        Set to enable code in RIOT that does safety checking which might not be
-        needed in a production environment but helps in the development
-        process.
+comment "Ideally, make a PR to add them to the Kconfig."
```
Comment (maribu) [inline Kconfig:39]: :heart:

**Review by maribu — APPROVED** [review]
Thx for addressing this!

**Comment by MrKevinWeiss** [comment]
maybe we can jump the queue on this one...

**Comment by MrKevinWeiss** [comment]
rebased

**Comment by MrKevinWeiss** [comment]
oh boy am I sad... OK back at it.

**Comment by MrKevinWeiss** [comment]
hmmm, way less of a problem than I though... just a conflict in values for `SDCARD_GENERIC_MTD_OFFSET`, I guess generally it should be 1?

**Comment by MrKevinWeiss** [comment]
Hmmm the failed test builds on my machine.  Maybe something to do with rust on a CI machine?

**Comment by maribu** [comment]
🎉 Thx!

**Comment by MrKevinWeiss** [comment]
I didn't think I would make it.  Nice that it is finally over.  I will also prioritize any issues or bugs this may have causes.

**Comment by benpicco** [comment]
Thank you for bringing the Kconfig transition to a close!

**Pull request merged**